File: KITE-Hangouts-Test/js/checks/AllVideoCheck.js
const {TestStep, Status, KiteTestError} = require('kite-common');
const TEN_SECONDS = 10000;

class AllVideoCheck extends TestStep {
  constructor(kiteBaseTest) {
    super();
    this.driver = kiteBaseTest.driver;
    this.numberOfParticipant = kiteBaseTest.numberOfParticipant;
    this.timeout = kiteBaseTest.timeout;
    this.page = kiteBaseTest.page;
    this.usersPerRoom = kiteBaseTest.payload.usersPerRoom;

    // Test reporter if you want to add attachment(s)
    this.testReporter = kiteBaseTest.reporter;
  }

  stepDescription() {
    return "Check the other videos are being received OK";
  }

  async step() {
    let indexArray = await this.page.getVideoIndex();
    if (indexArray.length === 0) {
      throw new KiteTestError(Status.FAILED, "There is no valid video on page");
    }
    if (indexArray.length < this.numberOfParticipant) {
      throw new KiteTestError(Status.FAILED, "There are not enough video to match the number of participants");

    }
    console.log("indexArray-> " + indexArray);
    let index;
    let result = "";
    let error = false;
    for (index = 0; index < indexArray.length; index ++) {
      let tmp = await this.page.videoCheck(indexArray[index], TEN_SECONDS);
      console.log('-> Received video[' + index + '] ' + tmp);
      result += tmp;
      if (index < indexArray.length - 1) {
        result += ' | ';
      }
      if (tmp !== 'video') {
        error = true;
      }
    }
    if (error) {
      this.testReporter.textAttachment(this.report, "Received videos", result, "plain");
      throw new KiteTestError(Status.FAILED, "Some videos are still or blank: " + result);
    }
  }
}

module.exports = AllVideoCheck;

File: KITE-Owt-Test/src/main/java/org/webrtc/kite/owt/checks/AllVideoCheck.java
package org.webrtc.kite.owt.checks;

import io.cosmosoftware.kite.exception.KiteTestException;
import io.cosmosoftware.kite.interfaces.Runner;
import io.cosmosoftware.kite.report.Status;
import io.cosmosoftware.kite.steps.TestStep;
import org.openqa.selenium.WebElement;
import org.webrtc.kite.owt.pages.OWTPage;

import java.util.List;

import static io.cosmosoftware.kite.entities.Timeouts.ONE_SECOND_INTERVAL;
import static io.cosmosoftware.kite.util.ReportUtils.getStackTrace;
import static io.cosmosoftware.kite.util.TestUtils.videoCheck;
import static io.cosmosoftware.kite.util.TestUtils.waitAround;

public class AllVideoCheck extends TestStep {

   
  private final int numberOfParticipants;
  private final OWTPage owtPage;


  public AllVideoCheck(Runner runner, int numberOfParticipants, OWTPage owtPage) {
    super(runner);
    this.numberOfParticipants = numberOfParticipants;
    this.owtPage = owtPage;
  }

  @Override
  public String stepDescription() {
    return "Check the other videos are being received OK";
  }

  @Override
  protected void step() throws KiteTestException {
    try {
      //wait a while to allow all videos to load.;
      List<WebElement> videos = owtPage.getVideoElements();
      int waitingTime = 0;
      while(videos.size() < numberOfParticipants && waitingTime < 10 * numberOfParticipants) {
        logger.debug("numberOfParticipants = " + numberOfParticipants + ", waitingTime = " + waitingTime);
        waitAround(ONE_SECOND_INTERVAL);
        videos = owtPage.getVideoElements();
        waitingTime += 1;
      }
      if (videos.size() < numberOfParticipants) {
        logger.error("videos.size() < numberOfParticipants");
        throw new KiteTestException(
          "Unable to find " + numberOfParticipants + " <video> element on the page. No video found = "
            + videos.size(), Status.FAILED);
      }
      waitAround(numberOfParticipants * ONE_SECOND_INTERVAL);
      String videoCheck = "";
      boolean error = false;
      for (int i = 1; i < numberOfParticipants; i++) {
        String v = videoCheck(webDriver, i);
        videoCheck += v;
        if (i < numberOfParticipants - 1) {
          videoCheck += "|";
        }
        if (!"video".equalsIgnoreCase(v)) {
          error = true;
        }
      }
      if (error) {
        reporter.textAttachment(report, "Received Videos", videoCheck, "plain");
        throw new KiteTestException("Some videos are still or blank: " + videoCheck, Status.FAILED);
      } else {
        logger.info("Video checks: " + videoCheck);
      }
    } catch (KiteTestException e) {
      logger.error(getStackTrace(e));
      throw e;
    } catch (Exception e) {
      logger.error(getStackTrace(e));
      throw new KiteTestException("Error looking for the video", Status.BROKEN, e);
    }
  }
}
